go/crosstool.rst

```diff
@@ -0,0 +1,340 @@
+    target_cpu: "x86_64"
+    target_libc: "x86_64"
+    compiler: "clang"
+    abi_version: "unknown"
+    abi_libc_version: "unknown"
+
+    tool_path { name: "ar" path: "bin/llvm-ar" }
+    tool_path { name: "cpp" path: "bin/clang-cpp" }
+    tool_path { name: "dwp" path: "bin/llvm-dwp" }
+    tool_path { name: "gcc" path: "bin/clang" }
+    tool_path { name: "gcov" path: "bin/llvm-profdata" }
+    tool_path { name: "ld" path: "bin/ld.lld" }
+    tool_path { name: "nm" path: "bin/llvm-nm" }
+    tool_path { name: "objcopy" path: "bin/llvm-objcopy" }
+    tool_path { name: "objdump" path: "bin/llvm-objdump" }
+    tool_path { name: "strip" path: "bin/llvm-strip" }
+
+    compiler_flag: "-no-canonical-prefixes"
+    linker_flag: "-no-canonical-prefixes"
+
+    compiler_flag: "-v"
+    cxx_builtin_include_directory: "/usr/include"
+  }
+
+  default_toolchain {
+    cpu: "x86_64"
+    toolchain_identifier: "clang"
+  }
+
+For a more complete example, build any ``cc_binary`` with Bazel without
+explicitly configuring ``CROSSTOOL``, then look at the ``CROSSTOOL`` that
+Bazel generates for the automatically detected host toolchain. This can
+be found in ``$(bazel info
+output_base)/external/bazel_tools/tools/cpp/CROSSTOOL``. (You have to build
+something with the host toolchain before this will show up).
+
+Some notes:
+
+* ``toolchain_identifier`` is the main name for the toolchain. You'll refer to
+  it using this identifier from other messages and from build files.
+* Most of the other fields at the top of ``toolchain`` are descriptive and
+  can have any value.
+* ``tool_path`` fields describe the various tools Bazel may invoke. The paths
+  are relative to the directory that contains the ``CROSSTOOL`` file.
+* ``compiler_flag`` and ``linker_flag`` are passed to the compiler and linker
+  on each invocation, respectively.
+* ``cxx_builtin_include_directory`` is a directory with include files that
+  the compiler may read. Without this declaration, these files won't be
+  visible in the sandbox. (TODO: make this hermetic).
+
+Step 4: Write a build file
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We'll create a set of targets that will link the CROSSTOOL into Bazel's
+toolchain system. It's likely this API will change in the future. This will be
+in ``tools/BUILD.bazel``.
+
+First, we'll create some ``filegroups`` that we can reference from other rules.
+
+.. code:: bzl
+
+  package(default_visibility = ["//visibility:public"])
+
+  filegroup(
+      name = "empty",
+      srcs = [],
+  )
+
+  filegroup(
+      name = "all",
+      srcs = glob([
+          "bin/*",
+          "lib/**",
+          "libexec/**",
+          "share/**",
+      ]),
+  )
+
+Next, we'll create a ``cc_toolchain`` target that tells Bazel where to find some
+important files. This API is undocumented and will very likely change in the
+future. We need to create one of these for each ``toolchain`` in ``CROSSTOOL``.
+The ``toolchain_identifier`` and ``cpu`` fields should match, and the
```

Comment:
maybe clarify that "should match" here means "should match the values in `CROSSTOOL`. At first I was confused becfause I thought that `toolchain_identifier` should be the same as `cpu`, which it's not.